src/lib/operators/get_table.cpp

```diff
@@ -44,18 +44,45 @@ void GetTable::_on_set_parameters(const std::unordered_map<ParameterID, AllTypeV
 
 std::shared_ptr<const Table> GetTable::_on_execute() {
   auto original_table = StorageManager::get().get_table(_name);
+
+  DebugAssert(this->transaction_context_is_set() ||
+                  [original_table]() {
+                    for (ChunkID chunk_id{0}; chunk_id < original_table->chunk_count(); ++chunk_id) {
+                      // Check, whether chunk got deleted either physically (removed from chunk-vector resp. memory).
+                      // or logically (fully invalidated)
+                      if (!original_table->get_chunk(chunk_id) ||
+                          original_table->get_chunk(chunk_id)->get_cleanup_commit_id() < MvccData::MAX_COMMIT_ID)
+                        return false;
+                    }
+                    return true;
+                  }(),
+              "GetTable needs to have a transaction context set to filter out deleted chunks.");
```

Comment:
"For tables with logically or physically deleted chunks, the transaction context must be set"?